autofs-5.0.5 - fix status privilege error

From: Ian Kent <raven@themaw.net>

The recent LSB init init script change prevent normal users from using
the status init script action. Maybe the (very poor) specification is
wrong or I misread it, in either case this behaviour is unacceptable.
---

 CHANGELOG             |    2 ++
 redhat/autofs.init.in |    2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)


--- autofs-5.0.5.orig/CHANGELOG
+++ autofs-5.0.5/CHANGELOG
@@ -47,6 +47,8 @@
 - fix error handing in do_mount_indirect().
 - expire thread use pending mutex.
 - remove ERR_remove_state() openssl call.
+- fix init script restart option.
+- fix init script status privilege error.
 
 03/09/2009 autofs-5.0.5
 -----------------------
--- autofs-5.0.5.orig/redhat/autofs.init.in
+++ autofs-5.0.5/redhat/autofs.init.in
@@ -154,7 +154,7 @@ function reload() {
 RETVAL=0
 
 # Only the root user may change the service status
-if [ `id -u` -ne 0 ]; then
+if [ `id -u` -ne 0 ] && [ "$1" != "status" ]; then
 	echo "insufficient privilege to change service status"
 	exit 4
 fi
